modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProvider.java

```diff
@@ -190,10 +193,44 @@ public Long getTotalViews(long companyId, String url)
 
 			TypeFactory typeFactory = _objectMapper.getTypeFactory();
 
-			return _objectMapper.readValue(
+			List<TrafficSource> trafficSources = _objectMapper.readValue(
 				response,
 				typeFactory.constructCollectionType(
 					List.class, TrafficSource.class));
+
+			Map<String, AcquisitionChannel> acquisitionChannels =
+				getAcquisitionChannels(companyId, url);
+
+			Collection<AcquisitionChannel> values =
+				acquisitionChannels.values();
+
+			Stream<AcquisitionChannel> stream = values.stream();
+
+			return stream.map(
+				acquisitionChannel -> {
+					Stream<TrafficSource> trafficSourcesStream =
```

Comment:
Does it worthy to use a map for the traffic sources so that the find op would be immediate and clearer?

Reply:
I think we won't save a lot of time, since we have a limited number of keys (2). I have decided to follow the same solution you proposed in the previous solution since create the map implies transforming the list into a map too.

Reply:
But of course, I can change it in order to make things clearer.

Reply:
That's exactly why I was proposing that, I was never proud of my first approach 😿

Reply:
Great!!, I have added this change to the pull